core/services/job/orm.go

```diff
@@ -845,42 +845,98 @@ func (o *orm) PipelineRunsByJobsIDs(ids []int32) (runs []pipeline.Run, err error
 	return runs, errors.Wrap(err, "PipelineRunsByJobsIDs failed")
 }
 
+func (o *orm) loadPipelineRunIDs(jobID *int32, offset, limit int, tx pg.Queryer) (ids []int64, err error) {
+	var filter string
+
+	if jobID != nil {
+		filter = "JOIN jobs USING(pipeline_spec_id) WHERE jobs.id = $1 AND "
+	} else {
+		filter = "WHERE "
+	}
+	stmt := fmt.Sprintf(`SELECT p.id FROM pipeline_runs AS p %s p.id >= $4 AND p.id <= $5
+			ORDER BY p.id DESC OFFSET $2 LIMIT $3`, filter)
+	var res sql.NullInt64
+
+	if err = tx.Get(&res, "SELECT MAX(id) FROM pipeline_runs"); err != nil {
+		err = errors.Wrap(err, "error while loading runs")
+		return
+	} else if !res.Valid {
+		// MAX() will return NULL if there are no rows in table.  This is not an error
+		return
+	}
+
+	maxID := res.Int64
+
+	// Only search the most recent n jobs (whether deleted or not), starting with n = 1000 and
```

Comment:
Why are we searching jobs? Isn't this loading pipeline run IDs for just one job?

Reply:
I should fix this comment, it should say "most recent n runs"